### DISCUSSION TITLE
how to add actuator to free joint？

### DISCUSSION THREAD

**Discussion by LHT-Liu** [google-deepmind/mujoco#1711] [Asking for Help]
I create a [free joint](https://mujoco.readthedocs.io/en/stable/XMLreference.html#body-joint) , however i don't know how to add [actuator ](https://mujoco.readthedocs.io/en/stable/XMLreference.html#actuator)for free joint,

Thanks for your help. @yuvaltassa

**Comment by yuvaltassa** [comment]
This should be an issue, not a discussion.

What kind of actuator do you want? You probably want to use [site transmission](https://mujoco.readthedocs.io/en/latest/computation/index.html#transmission) which lets you model rockets and also (unphysical) abstract Cartesian actuators.

**Reply by LHT-Liu** [reply]
My robot is a mobile manipulator, and I want to set up a free joint type of joint for the base, that is, they are not controlled by wheels, but as a whole. I need to ask how to set up the actuator in a free joint. Is there any problem with my way of thinking?

![image](https://github.com/google-deepmind/mujoco/assets/64388319/d43ba42b-02b4-40cb-9d74-8c6d25c8faad)


```
    <!-- 平移自由度驱动器 -->
    <motor name="motor_tx" joint="base_link_free_joint" axis="1 0 0" gear="1"/>
    <motor name="motor_ty" joint="base_link_free_joint" axis="0 1 0" gear="1"/>
    <motor name="motor_tz" joint="base_link_free_joint" axis="0 0 1" gear="1"/>
    
    <!-- 旋转自由度驱动器 -->
    <motor name="motor_rx" joint="base_link_free_joint" axis="1 0 0" gear="1" ctrllimited="true" ctrlrange="-1 1"/>
    <motor name="motor_ry" joint="base_link_free_joint" axis="0 1 0" gear="1" ctrllimited="true" ctrlrange="-1 1"/>
    <motor name="motor_rz" joint="base_link_free_joint" axis="0 0 1" gear="1" ctrllimited="true" ctrlrange="-1 1"/>
```
but there is an error
![image](https://github.com/google-deepmind/mujoco/assets/64388319/b25b3011-929e-4efd-8836-1ef69ba76187)

I want to know how to add actuator to a free joint. How should I change my previous code.

Thanks~

**Reply by yuvaltassa** [reply]
replace `axis="a b c" gear="1"` → `gear="a b c"` note that this will give you forces in the global frame. For local frame use site transmission. Also note that for rotations you want (e.g. around the z axis) `gear="0 0 0 0 0 1"`

**Reply by LHT-Liu** [reply]
> replace `axis="a b c" gear="1"` → `gear="a b c"` note that this will give you forces in the global frame. For local frame use site transmission. Also note that for rotations you want (e.g. around the z axis) `gear="0 0 0 0 0 1"`

thanks for your suggest, this does work，Is my code below correct?
```
  <actuator>

    <motor name="motor_tx" joint="base_link_free_joint" gear="1 0 0 0 0 0"/>
    <motor name="motor_ty" joint="base_link_free_joint" gear="0 1 0 0 0 0"/>
    <motor name="motor_tz" joint="base_link_free_joint" gear="0 0 1 0 0 0"/>
    
    <motor name="motor_rx" joint="base_link_free_joint" gear="0 0 0 1 0 0" ctrllimited="true" ctrlrange="-1 1"/>
    <motor name="motor_ry" joint="base_link_free_joint" gear="0 0 0 0 1 0" ctrllimited="true" ctrlrange="-1 1"/>
    <motor name="motor_rz" joint="base_link_free_joint" gear="0 0 0 0 0 1" ctrllimited="true" ctrlrange="-1 1"/>

    ......

  </actuator>
```
![image](https://github.com/google-deepmind/mujoco/assets/64388319/01e8d588-b1e8-492a-8bd4-5f40c611b801)

Why is there no free joint in the Joint on the right side of the software? Why does the robot not move when I drag the 6 actuators I defined in Control?

thanks @yuvaltassa

**Reply by yuvaltassa** [reply]
Free joints don't have sliders because of the quaternion. 
Try increasing the gear above 1 to 10 or 100
